src/main/java/com/Elessar/app/server/DirectMsgSender.java

```diff
@@ -16,14 +18,19 @@
  */
 public class DirectMsgSender implements MsgSender {
     private static final Logger logger = LogManager.getLogger(DirectMsgSender.class);
+    private final MetricManager metricManager;
     private final HttpClient httpClient;
 
-    public DirectMsgSender(HttpClient httpClient) {
+    public DirectMsgSender(HttpClient httpClient, MetricManager metricManager) {
         this.httpClient = httpClient;
+        this.metricManager = metricManager;
     }
 
     @Override
     public P2PMsgResponse send(List<Message> messages, String URL) throws Exception {
+        final Metric metric = metricManager.newMetric(new StringBuilder().append(MsgSender.MSG_SENDER).append(".")
```

Comment:
Should we change the metric name to something like "DirectMsgSender.send"?

Reply:
I thought of this before, but in database, it seems we didn't differentiate this metric name between different database

Reply:
why does this have anything to do with database?
all we want is to be able to distinguish direct msg sender vs kafka sender from metric log